src/background/permissions/permissions-ensurer.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+
+export class PermissionsEnsurer {
```

Comment:
Not sure how I feel about the name: why can't we go with PermissionGetter?

Ensuring we have permissions feels similar to getting permissions to me?

Reply:
For me, `PermissionGetter` doesn't make me expect that it would be requesting permissions so much as returning which permissions we already have. I like `PermissionEnsurer` or `PermissionRequester`

Reply:
or `BrowserPermissionsRequester` maybe

Reply:
I wasn't sure about the name either. Considering what this class does:
- check permissions (by injecting code)
- request permissions if needed
- interpreting error codes from chrome API
All that make me thing that is way more than a getter. Also, consider we may end up using this right before injecting our code into the target page. 
I can go with requester as technically speaking, we can't ensure we have permissions, in the sense of, the user may as well denied permissions to our extension.
I'm open to renaming here.